test/unit/strict-template-policy.html

```diff
@@ -0,0 +1,146 @@
+  <script>
+    Polymer = {
+      lazyRegister: true,
+      strictTemplatePolicy: true
+    };
+    // Errors thrown in custom element reactions are not thrown up
+    // the call stack to the dom methods that provoked them, so need
+    // to catch them here and prevent mocha from complaining about them
+    window.addEventListener('error', function(event) {
+      event.stopImmediatePropagation();
+      event.preventDefault();
+      if (!window.uncaughtError) {
+        window.uncaughtError = event;
+      }
+    });
+  </script>
+  <script src="../../../web-component-tester/browser.js"></script>
+  <link rel="import" href="../../polymer.html">
+</head>
+<body>
+
+  <dom-module id="trusted-element">
+    <template>Trusted</template>
+    <script>
+      HTMLImports.whenReady(function() {
+        Polymer({is: 'trusted-element'});
+      });
+    </script>
+  </dom-module>
+
+  <div id="target"></div>
+
+  <script>
+    suite('strictTemplatePolicy', function() {
+
+      teardown(function() {
+        window.uncaughtError = null;
+      });
+
+      // Errors thrown in custom element reactions are not thrown up
+      // the call stack to the dom methods that provoked them, so this
+      // wraps Chai's assert.throws to re-throw uncaught errors
+      function assertThrows(fn, re) {
+        assert.throws(function() {
+          fn();
+          // Throw any errors caught on window
+          if (window.uncaughtError) {
+            throw new Error(window.uncaughtError.message);
+          }
+          // Force polyfill reactions and/or async template stamping
+          Polymer.dom.flush();
+          // Throw any add'l errors caught on window
+          if (window.uncaughtError) {
+            throw new Error(window.uncaughtError.message);
+          }
+        }, re);
+      }
+
+      test('dom-bind', function() {
+        assertThrows(function() {
+          document.getElementById('target').innerHTML =
+          '<template is="dom-bind">' +
+          '  <div id="injected"></div>'+
+          '</template>`';
+        }, /dom-bind not allowed/);
+        assert.notOk(document.getElementById('injected'));
+      });
+
+      test('dom-if', function() {
+        assertThrows(function() {
+          document.getElementById('target').innerHTML =
+            '<template is="dom-if" if>' +
+            '  <div id="injected"></div>'+
+            '</template>`';
+        }, /template owner not trusted/);
+        assert.notOk(document.getElementById('injected'));
+      });
+
+      test('dom-repeat', function() {
+        assertThrows(function() {
+          document.getElementById('target').innerHTML =
+            '<template is="dom-repeat" items="[0]">' +
+            '  <div id="injected"></div>'+
+            '</template>`';
+        }, /template owner not trusted/);
+        assert.notOk(document.getElementById('injected'));
+      });
+
+      test('dom-module after registration', function() {
```

Comment:
Add a test that registers the same module name 3x.

Reply:
Added: https://github.com/Polymer/polymer/pull/5303/commits/1d529ab9512745457aa3873c6795624083658cc6#diff-e2ce14a791fe5ac94beadd7de8fcdc95R142